src/main/java/org/terasology/launcher/gui/javafx/CleanupController.java

```diff
@@ -0,0 +1,88 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.gui.javafx;
+
+import javafx.application.Platform;
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.Label;
+import javafx.stage.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.launcher.settings.AbstractLauncherSettings;
+import org.terasology.launcher.util.BundleUtils;
+import org.terasology.launcher.util.CleanupUtils;
+
+import java.io.IOException;
+
+public class CleanupController {
+
+    private static final Logger logger = LoggerFactory.getLogger(CleanupController.class);
+
+    private AbstractLauncherSettings launcherSettings;
+
+    @FXML
+    private Label descriptionLabel;
+    @FXML
+    private CheckBox deleteGameBox;
+    @FXML
+    private CheckBox deleteGameDataBox;
+    @FXML
+    private CheckBox deleteLauncherBox;
+    @FXML
+    private Button okButton;
+    @FXML
+    private Button cancelButton;
+
+    void initialize(final AbstractLauncherSettings newLauncherSettings, final Stage newStage) {
+        this.launcherSettings = newLauncherSettings;
+        setLabelStrings();
+    }
+
+    private void setLabelStrings() {
+        descriptionLabel.setText(BundleUtils.getLabel("cleanup_description"));
+        deleteGameBox.setText(BundleUtils.getLabel("cleanup_delete_game"));
+        deleteGameDataBox.setText(BundleUtils.getLabel("cleanup_delete_gameData"));
+        deleteLauncherBox.setText(BundleUtils.getLabel("cleanup_delete_launcher"));
+        okButton.setText(BundleUtils.getLabel("cleanup_ok"));
+        cancelButton.setText(BundleUtils.getLabel("cleanup_cancel"));
+    }
+
+    @FXML
+    public void doCleanupAction(ActionEvent event) throws IOException {  //TODO: catch and show a message instead of throwing
+        if (deleteGameBox.isSelected()) {
+            CleanupUtils.deleteGameDirectory(launcherSettings);
+        }
+        if (deleteGameDataBox.isSelected()) {
+            CleanupUtils.deleteGameDatairectory(launcherSettings);
+        }
+        if (deleteLauncherBox.isSelected()) {
+            CleanupUtils.deleteLauncherDirectory(launcherSettings);
+        }
+        Platform.exit();
+        System.exit(0);
+    }
+
+    @FXML
+    public void cancelCleanupAction(ActionEvent event) {
+        //TODO: is this the right way to close?
+        ((Node) event.getSource()).getScene().getWindow().hide();
```

Comment:
Okay, thanks for the clarification; I simply copied that code from the method used to close the Settings windows and I had some doubts since the name and the [documentation](https://docs.oracle.com/javase/8/javafx/api/javafx/stage/Window.html#hide--) seems quite misleading to me, since every time a window is opened [we create a new instance of the stage](https://github.com/MovingBlocks/TerasologyLauncher/pull/401/files/d416da3d71db8062cc68e2065379b81705b47e9b..55d90680a8cf1dcebd79832e51fb70c0e216c07c#diff-a9056608efd45f70476a138151ba8260R278), and then to close we "just" hide it. However, from [this answer](http://stackoverflow.com/a/27895501) seems that hiding actually allows it to be garbage collected.